test/runnable/testxmm.d

```diff
@@ -2388,6 +2388,31 @@ void test23009()
     assert(R.array == correct);
 }
 
+/*****************************************/
+// https://issues.dlang.org/show_bug.cgi?id=23077
+
+float test23077(float x)
+{
+    short i = *cast(short*)&x;
+    ++i;
+    return *cast(float*)&i; // this cast is not allowed in @safe code
+}
```

Comment:
There is also 23085 (likely related, but triggers an ICE in another location).


```suggestion
// https://issues.dlang.org/show_bug.cgi?id=23077

float test23077(float x)
{
    short i = *cast(short*)&x;
    ++i;
    return *cast(float*)&i; // this cast is not allowed in @safe code
}

// https://issues.dlang.org/show_bug.cgi?id=23085

float test23085(float x)
{
    byte i = *cast(byte*)&x;
    ++i;
    return *cast(float*)&i; // this cast is not allowed in @safe code
}
```

Reply:
I don't understand. Are these additional tests not working with this PR? Or they do work and you just want them added?

Reply:
It was causing an ICE in a different file (cod1 vs cgxmm), so it's not immediately obvious whether the fix covers both.